float64 posX
float64 posY
float64 posT
